src/pfe/portal/modules/LoadRunner.js

```diff
@@ -487,91 +505,99 @@ module.exports = class LoadRunner {
     * Socket event for connect
     * Logs connection success
     */
-    this.socket.on('connect', () => {
-      log.info('Loadrunner has connected')
-      this.up = true;
+    this.socket.on('connect', data => {
+      //if (data.projectID === this.project.projectID) {
+        log.info('Loadrunner has connected')
+        this.up = true;
+      //}
     });
 
     /**
     * Socket event for disconnect
     * Logs the error and starts reconnect procedure
     */
-    this.socket.on('disconnect', () => {
-      log.info('Loadrunner has disconnected')
-      // If this.up is false we're already trying to reconnect
-      if (this.up) {
-        // socket.io-client will automatically reconnect and trigger the connect event.
-        this.up = false;
+    this.socket.on('disconnect', data => {
+      if (data.projectID === this.project.projectID) {
+        log.info('Loadrunner has disconnected')
+        // If this.up is false we're already trying to reconnect
+        if (this.up) {
+          // socket.io-client will automatically reconnect and trigger the connect event.
+          this.up = false;
+        }
       }
     });
 
     /**
     * Socket event for connection timeout
     * Logs the error and starts reconnect procedure
     */
-    this.socket.on('connect_timeout', () => {
-      log.info('Loadrunner has timed out')
-      // If this.up is false we're already trying to reconnect
-      if (this.up) {
-        // socket.io-client will automatically reconnect and trigger the connect event.
-        this.up = false;
-      }
+    this.socket.on('connect_timeout', data => {
+      if (data.projectID === this.project.projectID) {
+        log.info('Loadrunner has timed out')
+        // If this.up is false we're already trying to reconnect
+        if (this.up) {
+          // socket.io-client will automatically reconnect and trigger the connect event.
+          this.up = false;
+        }
+      }  
     });
 
     /**
     * Socket event for error
     * Logs the error and starts reconnect procedure
     */
-    this.socket.on('error', (err) => {
-      log.error('LoadRunner socket error');
-      // A socket error could occur while a load run is not in progress.
-      if (this.project) {
-        this.project.loadInProgress = false;
-      }
-      log.error(err);
-      // If this.up is false we're already trying to reconnect
-      if (this.up) {
-        // socket.io-client will automatically reconnect and trigger the connect event.
-        this.up = false;
+    this.socket.on('error', (err, data) => {
```

Comment:
On there's no call to `this.setStatus` in the error handler here to reset state. The old loadInProgress flag is set to false, do you need to set the run status here too?